guides/ssb-server-context.md

```diff
@@ -0,0 +1,36 @@
+
+> Sometimes, ssb-server is packaged as an "embedded" process with an SSB application, such as Patchwork. Two or more versions of ssb-server connecting to the same identity and feed (log) should not run at the same time. The SSB ecosystem is currently exploring a variety of solutions to this problem.
+
+There are a variety of ways to access the API functions of ssb-server.
+
+If the ssb-server process is started using the ssb-server command line command, it can be accessed:
+- via a separate/second command line interface, using other ssb-server command line commands
+- via ssb-client in a nodejs program, which connects to it via RPC (remote procedure calls) and has complete access to the API including extended functionality from plugins
+
+If the ssb-server process is started programmatically, by importing it into a nodejs file, it can be accessed:
+- directly via the in-program reference to the running server
+- via a separate/second command line interface, using other ssb-server command line commands
+- via ssb-client in a nodejs program, which connects to it via RPC (remote procedure calls) and has complete access to the API including extended functionality from plugins
+
+## What is ssb-client?
+[ssb-client](https://github.com/ssbc/ssb-client) is a nodejs module that can connect to a running `ssb-server` instance and expose its entire API, including that of the extended plugins, to a nodejs program. This resolves the concern over multiple `ssb-server` instances running, as one `ssb-server` process can be run, and one or more instances of `ssb-client` can connect to it.
+
+In many cases, `ssb-client` will be a good selection for interacting with a running `ssb-server` process.
+
+## Accessing API functions
+
+The following subarticles of this guide are divided according to core functionality of a running ssb-server process that you may wish to utilize, such as reading from, and publishing to, the feed (append-only log) for an idenity.
+
+In each article, it is typical for there to be two references on how to access that function:
+
+1. via the command line
+2. via a reference in a program to the `ssb-server` API.
+
+For the second one, this is written as `sbot`, and it is important to note that this `sbot` instance could be an original reference to a programmatically started `ssb-server` instance, or it could be an instance of an `ssb-client` connected to a running `ssb-server` process. Note that either way, the API is the same, and so we can call it a generic `sbot`, short for Scuttlebot.
```

Comment:
gotcha : the API is subtly different. If you are directly interacting with `ssb-server` the "sync" methods are in fact synchronous (e.g. `var data = sbot.whoami()`), but if you're connecting over RPC, then "sync" methods are converted to "async" versions!  (e.g. `sbot.whoami((err, data) => { ...  })`).

This isn't common but it is important to know about

Reply:
oo, great, thankyou